gammapy/background/energy_offset_array.py

```diff
@@ -71,7 +95,11 @@ def _fill_one_event_list(self, events):
         ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
                                                      [self.energy.value,
                                                       self.offset.value])
-        return ev_cube_hist
+        energy_bin = np.diff(ev_cube_edges[0])
+        solid_angle_bin = np.pi * np.diff(ev_cube_edges[1] ** 2)
+        solid_angle_tab, energy_tab = np.meshgrid(solid_angle_bin, energy_bin)
+        bkg_rate = ev_cube_hist / (solid_angle_tab * energy_tab)
```

Comment:
Do you really think that we have to do this in two separated PR because there are linked to each other?
I would prefer to do both in the same PR? 
Le 9 févr. 2016 à 18:34, Lea Jouvin <ljouvin@apc.in2p3.fr> a écrit :

> Yes I see like the Cube and CubeBackgroundModel. I will try to look at it in more details.
> I am not here tomorrow since I have a formation but I will start this on Thursday.
> Le 9 févr. 2016 à 17:31, Christoph Deil <notifications@github.com> a écrit :
> 
>> In gammapy/background/energy_offset_array.py:
>> 
>> > @@ -71,7 +95,11 @@ def _fill_one_event_list(self, events):
>> >          ev_cube_hist, ev_cube_edges = np.histogramdd(ev_cube_array,
>> >                                                       [self.energy.value,
>> >                                                        self.offset.value])
>> > -        return ev_cube_hist
>> > +        energy_bin = np.diff(ev_cube_edges[0])
>> > +        solid_angle_bin = np.pi * np.diff(ev_cube_edges[1] ** 2)
>> > +        solid_angle_tab, energy_tab = np.meshgrid(solid_angle_bin, energy_bin)
>> > +        bkg_rate = ev_cube_hist / (solid_angle_tab * energy_tab)
>> I think both the EnergyOffsetArray class and the container class that contains three arrays should have read / write.
>> I.e. keep the read / write here as-is!
>> 
>> The container class then calls the to_fits for each array and puts the three HDUs in one FITS file with extension names like "counts", "livetime" and "rate" (or whatever the 3D cube class does).
>> 
>> Does this make sense?
>> 
>> (You could already start the pull request for the container class now, and for now just do the rate computation there and I/O, and then we add other stuff as needed in the coming days ...)
>> 
>> —
>> Reply to this email directly or view it on GitHub.
>> 
>